templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,114 @@
+    PORT=$(nmcli -t -f connection.master conn show "$INTERFACE_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the interface's NM uuid
+    PORT_CONNECTION_UUID=$(nmcli -t -f  device,type,uuid conn | awk -F ':' '{if( ($1=="'${PORT}'" || $3=="'${PORT}'") && $2~/^ovs*/) print $NF}')
+    if [ "$PORT_CONNECTION_UUID" == "" ]; then
+        exit 0
+    fi
+    
+    # Get the port's slave-type. If this is not an ovs-bridge, then exit
+    PORT_OVS_SLAVE_TYPE=$(nmcli -t -f connection.slave-type conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    if [ "$PORT_OVS_SLAVE_TYPE" != "ovs-bridge" ]; then
+        exit 0
+    fi
+    
+    # Get the port's master = bridge name
+    BRIDGE_NAME=$(nmcli -t -f connection.master conn show "$PORT_CONNECTION_UUID" | awk -F ':' '{print $NF}')
+    # Limit this to br-ex and br-ex1 only. If one wanted to enable this for all OVS bridges,
+    # the condition would be: if [ "$BRIDGE_NAME" == "" ]; then
+    if [ "$BRIDGE_NAME" != "br-ex" ] && [ "$BRIDGE_NAME" != "br-ex1" ]; then
+        exit 0
+    fi
+    
+    CONFIGURATION_FILE=/run/ofport_requests.${BRIDGE_NAME}
+    
+    # Declare a new map. If CONFIGURATION_FILE exists, source entries from there
+    declare -A INTERFACES
+    if [ -f $CONFIGURATION_FILE ]; then
+        echo "Sourcing configuration file '${CONFIGURATION_FILE}' with contents: $(cat $CONFIGURATION_FILE)"
+        source $CONFIGURATION_FILE
+    fi
+    
+    # get_interface_ofport_request will return
+    # * either: the current ofport assignment for the port if no interface has claimed this ofport number, yet
+    # * or:     the lowest available free ofport number
+    function get_interface_ofport_request() {
+        # Build a map that only contains the currently reserved ofport_requests
+        declare -A ofport_requests
+        for interface_name in "${!INTERFACES[@]}"; do
+    	ofport_requests[${INTERFACES[$interface_name]}]=${INTERFACES[$interface_name]}
+        done
+    
+        # Get the current ofport number assignment
+        local current_ofport=$(ovs-vsctl get Interface "$INTERFACE_NAME" ofport)
+    
+        # If the current ofport number is still free, use it
+        if ! [ "${ofport_requests[$current_ofport]+a}" ]; then
+    	    echo $current_ofport
+    	    return
+        fi
+    
+        # If the current ofport number is not free, return the lowest free entry
+        i=0
+        for i in {1..65000}; do
+            if ! [ "${ofport_requests[$i]+a}" ]; then
+                echo $i
+                return
+            fi
+        done
+    
+        # if we still cannot find an ID, exit with an error
+        echo "Impossible to find an ofport ID for interface $INTERFACE_NAME" >&2
+        exit 1
+    }
+    
+    # If INTERFACES[INTERFACE_NAME] exists, use that value
+    # If INTERFACES[INTERFACE_NAME] does not exists, use get_interface_ofport_request to get a new value
```

Comment:
If one doesn't exist and the port is already created in OVS why don't you just get the current one assigned by OVS if there isn't a saved value already for this port?

Reply:
I'm doing that already:
~~~
  72     # get_interface_ofport_request will return
 73     # * either: the current ofport assignment for the port if no interface has claimed this ofport number, yet
 74     # * or:     the lowest available free ofport number
 75     function get_interface_ofport_request() {
 76         # Build a map that only contains the currently reserved ofport_requests
 77         declare -A ofport_requests
 78         for interface_name in "${!INTERFACES[@]}"; do
 79             ofport_requests[${INTERFACES[$interface_name]}]=${INTERFACES[$interface_name]}
 80         done
 81 
 82         # Get the current ofport number assignment
 83         local current_ofport=$(ovs-vsctl get Interface "$INTERFACE_NAME" ofport)
 84 
 85         # If the current ofport number is still free, use it
 86         if ! [ "${ofport_requests[$current_ofport]+a}" ]; then
 87             echo $current_ofport
 88             return
 89         fi
(...)
~~~

I'll adjust the highlighted comment to make sure that this is clearer.